File: javascript/grammer/11_sort_key_lambda.js
const students = [['kim', 70], ['lee', 90], ['park', 80]];

const byScore = [...students].sort((a, b) => a[1] - b[1]);
const byNameDesc = [...students].sort((a, b) => (a[0] < b[0] ? 1 : -1));

console.log('원본:', students);
console.log('점수순:', byScore);
console.log('이름 내림차순:', byNameDesc);

File: python/grammer/11_sort_key_lambda.py
"""11) 정렬: key와 lambda로 기준 지정."""

students = [("kim", 70), ("lee", 90), ("park", 80)]

# 점수 기준 오름차순
by_score = sorted(students, key=lambda x: x[1])
# 이름 기준 내림차순
by_name_desc = sorted(students, key=lambda x: x[0], reverse=True)

print("원본:", students)
print("점수순:", by_score)
print("이름 내림차순:", by_name_desc)
